db-engine/src/main/java/com/splicemachine/db/impl/jdbc/authentication/AuthenticationServiceBase.java

```diff
@@ -137,12 +139,38 @@
     */
     protected static final int SECMEC_USRSSBPWD = 8;
 
+	private static final String IMPERSIONATION_ENABLED = "derby.authentication.impersonation.enabled";
+	private static final String IMPERSIONATION_USERS = "derby.authentication.impersonation.users";
+
 	//
 	// constructor
 	//
 	public AuthenticationServiceBase() {
 	}
 
+	protected String impersonate(String userName, String proxyUser) {
+		if (!Boolean.parseBoolean(getProperty(IMPERSIONATION_ENABLED))) {
+			 return null;
+		}
+
+		String allPermissions = getProperty(IMPERSIONATION_USERS);
+
+		Map<String, String> map = Splitter.on(';').withKeyValueSeparator("=").split(allPermissions);
+		String userPermissions = map.get(userName);
+
+		if (userPermissions == null) {
+			return null;
+		}
+
+		Iterable<String> perms = Splitter.on(',').split(userPermissions);
+		for (String p : perms) {
+			if (p.equals("*") || p.equals(proxyUser))
```

Comment:
A minor comment is whether we need to compare ignoring case. Otherwise, it seems that the username passed in through the jdbc connection would be sensitive to case.